### PR TITLE
GEOPY-715

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-715 - Prototype joint inversion driver (single property)**

### CODE REVIEW

File: geoh5py/ui_json/input_file.py [hunk 1]
```diff
@@ -367,6 +385,37 @@ def write_ui_json(
 
         return self.path_name
 
+    def set_data_value(self, data: dict):
```
Comment (benk-mira) [inline geoh5py/ui_json/input_file.py:416]: The only place we use this is in the setter_validator on params where it handles a single value at a time.  Maybe just pass key, value in from the caller and ditch the loop?

File: geoh5py/ui_json/input_file.py [hunk 2]
```diff
@@ -270,16 +271,58 @@ def update_ui_values(self, data: dict):
...
+        if not isinstance(value, bool):
+            raise ValueError("Input value for `validate` should be True or False.")
+
+        self._validate = value
+
     @property
     def validation_options(self):
-        """Pass validation options to the validators."""
+        """
+        Pass validation options to the validators. The following parameters are
```
Comment (benk-mira) [inline geoh5py/ui_json/input_file.py:289]: Split the lines.  Should be a one liner followed by a new paragraph for details like the supported params

File: geoh5py/ui_json/input_file.py [hunk 3]
```diff
@@ -270,16 +271,58 @@ def update_ui_values(self, data: dict):
...
+        supporter:
```
Comment (benk-mira) [inline geoh5py/ui_json/input_file.py:290]:
```suggestion
        supported:
```

File: geoh5py/ui_json/input_file.py (hunk 2 above)
Comment (benk-mira) [inline geoh5py/ui_json/input_file.py:289]:
```suggestion
        Pass validation options to the validators. The following options are
```

File: geoh5py/ui_json/input_file.py [hunk 4]
```diff
@@ -270,16 +271,58 @@ def update_ui_values(self, data: dict):
...
         return self._validation_options
 
+    @validation_options.setter
+    def validation_options(self, value: dict):
+        if not isinstance(value, (dict, type(None))):
+            raise ValueError("Input value for `validation_options` should be a dict.")
+
+        if value is not None:
+            for key in value:
+                if key not in self.validation_options:
```
Comment (benk-mira) [inline geoh5py/ui_json/input_file.py:314]: Validating the validations - Nice!

File: geoh5py/ui_json/validation.py [hunk 5]
```diff
@@ -264,7 +264,8 @@ def validate(
 
     def validate_data(self, data: dict[str, Any]) -> None:
         """
-        Calls validate method on individual key/value pairs in input.
+        Calls validate method on ui.json data structure for cross-dependencies,
```
Comment (benk-mira) [inline geoh5py/ui_json/validation.py:253]: Should be a one-liner followed by more details

**Review by benk-mira — COMMENTED** [review]
A few minor comments

File: geoh5py/ui_json/input_file.py (hunk 1 above)
Comment (domfournier) [reply geoh5py/ui_json/input_file.py:416]: Good point. Done

File: geoh5py/ui_json/input_file.py (hunk 2 above)
Comment (domfournier) [reply geoh5py/ui_json/input_file.py:289]: Done

Comment (domfournier) [reply geoh5py/ui_json/input_file.py:289]: Yip

File: geoh5py/ui_json/validation.py (hunk 5 above)
Comment (domfournier) [reply geoh5py/ui_json/validation.py:253]: Done

**Review by benk-mira — COMMENTED** [review]
Thanks for the updates!

**Pull request merged**